modules/ocf_mirrors/manifests/monitoring.pp

```diff
@@ -6,19 +6,27 @@
     $local_path = "/${title}",
     $upstream_path = "/${title}",
     $upstream_protocol = 'http',
+    $ts_path = undef,
     $ensure = 'present',
   ) {
-  $local_url = "http://mirrors.ocf.berkeley.edu${local_path}/dists/${dist_to_check}/Release"
-  $upstream_url = "${upstream_protocol}://${upstream_host}${upstream_path}/dists/${dist_to_check}/Release"
+
+  if $type == 'debian' {
+    $local_url = "http://mirrors.ocf.berkeley.edu${local_path}/dists/${dist_to_check}/Release"
+    $upstream_url = "${upstream_protocol}://${upstream_host}${upstream_path}/dists/${dist_to_check}/Release"
+  }
+  elsif $type == 'ts' {
+    $local_url ="http://mirrors.ocf.berkeley.edu${local_path}/${ts_path}"
+    $upstream_url = "${upstream_protocol}://${upstream_host}${upstream_path}/${ts_path}"
```

Comment:
Since the first part of our mirror and the upstream mirror is the same, maybe put those in variables above the condition? I think that'll help make it a but clearer, especially if we need to change something for both types of health checks at once (or, more likely, add a third type of health check)